applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -0,0 +1,171 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Simon Wenczowski
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/openmp_utils.h"
+#include "processes/find_nodal_h_process.h"
+
+// Application includes
+#include "mass_conservation_check_process.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+MassConservationCheckProcess::MassConservationCheckProcess(
+        ModelPart& rModelPart,
+        const int MassComputationFreq,
+        const bool CompareToInitial,
+        const bool WriteToLogFile)
+    : Process(), mrModelPart(rModelPart) {
+
+    mMassComputationFreq = MassComputationFreq;
+    mCompareToInitial = CompareToInitial;
+    mWriteToLogFile = WriteToLogFile;
+}
+
+
+MassConservationCheckProcess::MassConservationCheckProcess(
+    ModelPart& rModelPart,
+    Parameters& rParameters)
+    : Process(), mrModelPart(rModelPart) {
+
+    Parameters default_parameters( R"(
+    {
+        "model_part_name"                        : "default_model_part_name",
+        "mass_computation_frequency"             : 20,
+        "compare_to_initial_values"              : true,
+        "write_to_log_file"                      : true
+    }  )" );
+
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+
+    mMassComputationFreq = rParameters["mass_computation_frequency"].GetInt();
+    mCompareToInitial = rParameters["compare_to_initial_values"].GetBool();
+    mWriteToLogFile = rParameters["write_to_log_file"].GetBool();
+}
+
+
+bool MassConservationCheckProcess::GetUpdateStatus(){
```

Comment:
Ok. I will move this to an function named "Execute". The computation of inflow/outflow was already implemented is currently in a different branch. It would be possible to add it here and, thus, to avoid a separate pull request. What would you prefer?

Reply:
Add it in a separate PR.

Reply:
Nice! I guess it is better to add it in a separate PR. Good work by the way!

Reply:
Function "Execute" was added in the python script of the process. As I personally experienced, it is dangerous to return multiple values from Cpp to python (or maybe I just did not find the right way to do so). I still hope that it solves the problem and gives the process a "compact" execute.

Reply:
Ok, I undestand your problem now. I do not personally like it as I think every cpp process itself should have an Execute method. If for any reason it cannot be done as in this case then it should be an utility instead of a process. Been said this, whatever you and @rubenzorrilla decided is okay to me..

Reply:
> I understand this is not part of the current PR, but it would be nice if this could take into account inflow volume, but its just a comment...

@ddiezrod This is now addressed in #3674